client/state/application-passwords/test/reducer.js

```diff
@@ -8,73 +8,111 @@ import deepFreeze from 'deep-freeze';
-		{
-			ID: 12345,
-			name: 'Example',
-			value: '2018-01-01T00:00:00+00:00',
-		},
-		{
-			ID: 23456,
-			name: 'Test',
-			value: '2018-02-01T08:10:00+00:00',
-		},
-	];
-	const otherAppPasswords = [
-		{
-			ID: 54321,
-			name: 'Sample',
-			value: '2018-03-01T05:30:00+00:00',
-		},
-	];
-
-	test( 'should default to an empty array', () => {
+	test( 'should export expected reducer keys', () => {
 		const state = reducer( undefined, {} );
-		expect( state ).toEqual( [] );
+
+		expect( state ).toHaveProperty( 'items' );
+		expect( state ).toHaveProperty( 'newPassword' );
 	} );
 
-	test( 'should set application passwords to empty array when user has no application passwords', () => {
-		const state = reducer( undefined, {
-			type: APPLICATION_PASSWORDS_RECEIVE,
-			appPasswords: [],
+	describe( 'items', () => {
+		const appPasswords = [
+			{
+				ID: 12345,
+				name: 'Example',
+				value: '2018-01-01T00:00:00+00:00',
+			},
+			{
+				ID: 23456,
+				name: 'Test',
+				value: '2018-02-01T08:10:00+00:00',
+			},
+		];
+		const otherAppPasswords = [
+			{
+				ID: 54321,
+				name: 'Sample',
+				value: '2018-03-01T05:30:00+00:00',
+			},
+		];
+
+		test( 'should default to an empty array', () => {
+			const state = items( undefined, {} );
+			expect( state ).toEqual( [] );
 		} );
 
-		expect( state ).toEqual( [] );
-	} );
+		test( 'should set application passwords to empty array when user has no application passwords', () => {
+			const state = items( undefined, {
+				type: APPLICATION_PASSWORDS_RECEIVE,
+				appPasswords: [],
+			} );
 
-	test( 'should add application passwords to the initial state', () => {
-		const state = reducer( [], {
-			type: APPLICATION_PASSWORDS_RECEIVE,
-			appPasswords,
+			expect( state ).toEqual( [] );
 		} );
 
-		expect( state ).toEqual( appPasswords );
-	} );
+		test( 'should add application passwords to the initial state', () => {
+			const state = items( [], {
+				type: APPLICATION_PASSWORDS_RECEIVE,
+				appPasswords,
+			} );
 
-	test( 'should overwrite previous application passwords in state', () => {
-		const state = deepFreeze( appPasswords );
-		const newState = reducer( state, {
-			type: APPLICATION_PASSWORDS_RECEIVE,
-			appPasswords: otherAppPasswords,
+			expect( state ).toEqual( appPasswords );
 		} );
 
-		expect( newState ).toEqual( otherAppPasswords );
+		test( 'should overwrite previous application passwords in state', () => {
+			const state = deepFreeze( appPasswords );
+			const newState = items( state, {
+				type: APPLICATION_PASSWORDS_RECEIVE,
+				appPasswords: otherAppPasswords,
+			} );
+
+			expect( newState ).toEqual( otherAppPasswords );
+		} );
+
+		test( 'should delete application passwords by ID from state', () => {
+			const state = deepFreeze( appPasswords );
+			const newState = items( state, {
+				type: APPLICATION_PASSWORD_DELETE_SUCCESS,
+				appPasswordId: appPasswords[ 0 ].ID,
+			} );
+
+			expect( newState ).toEqual( [ appPasswords[ 1 ] ] );
```

Comment:
Would this be clearer and more resilient with [`reject`](https://lodash.com/docs/4.17.5#reject)?

Reply:
I don't think it will be clearer, plus we'd have to use something different for the new state too; and concerning how resilient it would be - I don't mind tests being less resilient; after all - if this breaks, then we probably need to fix the method, and not the test. Leaving as-is 😉